scipy/stats/tests/test_stats.py

```diff
@@ -7663,6 +7663,26 @@ def test_error_code(self):
             v_values = rng.random(size=(2, 2))
             _ = stats.wasserstein_distance(u_values, v_values)
 
+    @pytest.mark.parametrize('u_size', [1, 10, 300])
+    @pytest.mark.parametrize('v_size', [1, 10, 300])
+    def test_optimization_vs_analytical(self, u_size, v_size):
+        rng = np.random.default_rng(45634745675)
+        # Test when u_weights = None, v_weights = None
+        u_values = rng.random(size=(u_size, 1))
+        v_values = rng.random(size=(v_size, 1))
+        u_values_flat = u_values.ravel()
+        v_values_flat = v_values.ravel()
+        # These two calculations are done using different backends
+        # but they must be equal
+        assert_allclose(stats.wasserstein_distance(u_values, v_values),
+                stats.wasserstein_distance(u_values_flat, v_values_flat))
+        # Test with u_weights and v_weights specified.
+        u_weights = rng.random(size=u_size)
+        v_weights = rng.random(size=v_size)
+        assert_allclose(
+        stats.wasserstein_distance(u_values, v_values, u_weights, v_weights),
+        stats.wasserstein_distance(u_values_flat, v_values_flat, u_weights, v_weights))
```

Comment:
Let's see what CI says, but I think PEP8 would ask for four spaces indentation within the `assert_allclose` (not eight or zero).

Otherwise, looks good.

Reply:
Lint looks okay.

Reply:
This is very much a nit but I do find this quite hard to read. May I suggest 
```suggestion
        assert_allclose(stats.wasserstein_distance(u_values, v_values),
                        stats.wasserstein_distance(u_values_flat, v_values_flat))
        # Test with u_weights and v_weights specified.
        u_weights = rng.random(size=u_size)
        v_weights = rng.random(size=v_size)
        d1 = stats.wasserstein_distance(u_values, v_values,
                                        u_weights, v_weights)
        d2 = stats.wasserstein_distance(u_values_flat, v_values_flat,
                                        u_weights, v_weights))
        assert_allclose(d1, d2)
```

Reply:
Thank you @lucascolley. Currently, @mdhaber has a pull request which reverts the changes that made these tests necessary. I am waiting on that.